config/size-report.js

```diff
@@ -0,0 +1,217 @@
+  const jobUrl = buildData.builds[0].jobs[0]['@href'];
+  const log = await getTravisText(jobUrl + '/log.txt');
+  const reResult = buildSizePrefixRe.exec(log);
+
+  if (!reResult) return;
+  return JSON.parse(reResult[1]);
+}
+
+/**
+ * Generate an array that represents the difference between builds.
+ * Returns an array of { beforeName, afterName, beforeSize, afterSize }.
+ * Sizes are gzipped size.
+ * Before/after properties are missing if resource isn't in the previous/new build.
+ */
+function getChanges(previousBuildInfo, buildInfo) {
+  const buildChanges = [];
+  const alsoInPreviousBuild = new Set();
+
+  for (const oldEntry of previousBuildInfo) {
+    const newEntry = buildInfo.find(entry => entry.name === oldEntry.name) ||
+      findHashedMatch(oldEntry.name, buildInfo);
+
+    // Entry is in previous build, but not the new build.
+    if (!newEntry) {
+      buildChanges.push({
+        beforeName: oldEntry.name,
+        beforeSize: oldEntry.gzipSize,
+      });
+      continue;
+    }
+
+    // Mark this entry so we know we've dealt with it.
+    alsoInPreviousBuild.add(newEntry);
+
+    // If they're the same, just ignore.
+    // Using size rather than gzip size. I've seen different platforms produce different zipped
+    // sizes.
+    if (
+      oldEntry.size === newEntry.size &&
+      oldEntry.name === newEntry.name
+    ) continue;
+
+    // Entry is in both builds (maybe renamed).
+    buildChanges.push({
+      beforeName: oldEntry.name,
+      afterName: newEntry.name,
+      beforeSize: oldEntry.gzipSize,
+      afterSize: newEntry.gzipSize,
+    });
+  }
+
+  // Look for entries that are only in the new build.
+  for (const newEntry of buildInfo) {
+    if (alsoInPreviousBuild.has(newEntry)) continue;
+
+    buildChanges.push({
+      afterName: newEntry.name,
+      afterSize: newEntry.gzipSize,
+    });
+  }
+
+  return buildChanges;
+}
+
+async function main() {
+  // Output the current build sizes for later retrieval.
+  const buildInfo = await dirToInfoArray(__dirname + '/../build');
+  console.log(buildSizePrefix + JSON.stringify(buildInfo));
+  console.log('\nBuild change report:');
+
+  let previousBuildInfo;
+
+  try {
+    previousBuildInfo = await getPreviousBuildInfo();
+  } catch (err) {
+    console.log(`  Couldn't parse previous build info`);
+    return;
+  }
+
+  if (!previousBuildInfo) {
+    console.log(`  Couldn't find previous build info`);
+    return;
+  }
+
+  const buildChanges = getChanges(buildInfo, previousBuildInfo);
+
+  if (buildChanges.length === 0) {
+    console.log('  No changes');
+    return;
+  }
+
+  // One letter references, so it's easier to get the spacing right.
+  const y = chalk.yellow;
+  const g = chalk.green;
+  const r = chalk.red;
+
+  for (const change of buildChanges) {
+    // New file.
+    if (!change.beforeSize) {
+      console.log(`  ${g('ADDED')}   ${change.afterName} - ${prettyBytes(change.afterSize)}`);
```

Comment:
Have you tried `console.group()` to see how Travis handles that?

Reply:
Travis appears to use markers for folding https://github.com/travis-ci/travis-ci/issues/2285#issuecomment-42724719, but it isn't clear how to unfold.

Reply:
Maybe shout into the Twitter void?